### PR TITLE
[txn-emitter] Add WaveTps mode, for creating a sine-wave-like workload pattern

### PR DESCRIPTION
**Pull request by igor-aptos**

### CODE REVIEW

File: crates/transaction-emitter-lib/src/emitter/submission_worker.rs [hunk 1]
```diff
@@ -62,21 +62,17 @@ impl SubmissionWorker {
...
     pub(crate) async fn run(mut self) -> Vec<LocalAccount> {
-        // Introduce a random jitter between, so that:
-        //  - we don't hammer the rest APIs all at once.
-        //  - allow for even spread for fixed TPS setup
-        let start_sleep_duration = self.start_sleep_time();
-        let start_time = Instant::now() + start_sleep_duration;
+        let start_time = Instant::now() + self.start_sleep_duration;
 
-        self.sleep_check_done(start_sleep_duration).await;
+        self.sleep_check_done(self.start_sleep_duration).await;
```
Comment (bchocho) [inline crates/transaction-emitter-lib/src/emitter/submission_worker.rs:75]: I'm obviously missing something. How does this alter the wait time between requests, and not just the start wait time?

File: crates/transaction-emitter-lib/src/emitter/mod.rs [hunk 2]
```diff
@@ -411,6 +442,65 @@ impl EmitJobRequest {
...
+                        / (self.workers_per_endpoint * self.endpoints) as f64;
+                    (start_offset_multiplier_millis * i as f64) as u64
+                })
+                .collect(),
+            WorkerOffsetMode::Wave {
+                wave_ratio,
+                num_waves,
+            } => {
+                // integral (1 + wave_ratio cos((2PI num_waves x)/wait_millis)) dx =
+                // (x - (wave_ratio wait_millis sin( (num_waves 2PI x)/wait_millis ))  /  (num_waves 2PI))
```
[the remark this answers is not in the document]
Comment (msmouse) [reply crates/transaction-emitter-lib/src/emitter/mod.rs:471]: I see.. update the comment please :D

File: crates/transaction-emitter-lib/src/emitter/submission_worker.rs (hunk 1 above)
Comment (igor-aptos) [reply crates/transaction-emitter-lib/src/emitter/submission_worker.rs:75]: it only affects the start wait time. 
It creates a pattern within the "wait_millis", that then repeats every "wait_millis" seconds

**Review by bchocho — APPROVED** [review]
Everything but the math 🙈

**Review by msmouse — APPROVED** [review]
math seems sound to me

**Pull request merged**